platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheProblemReportingIntegrationTest.groovy

```diff
@@ -272,6 +273,58 @@ class ConfigurationCacheProblemReportingIntegrationTest extends AbstractConfigur
         }
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/13862")
+    def "state deserialization errors always halt the build and includes the cause"() {
```

Comment:
💅 
```suggestion
    def "state deserialization errors fail the build and include the cause"() {
```

Reply:
The "always" bit is relevant though (meaning - on first and second run). Note that I basically replicated the existing test that checked handling of serialization errors, but tweaked it to cover deserialization instead.

Reply:
I was only looking at the diff, so didn't notice. Makes sense 👍